common.c

```diff
@@ -563,7 +567,7 @@ void log_message(int type, char* msg, ...)
 }
 
 /* syslogs who connected to where */
-void log_connection(struct connection *cnx)
+void log_connection(struct connection *cnx, int do_proctitle)
```

Comment:
I'd argue that setting the process name is also "kind of" logging, but you are right, it should be a different function.

The main reason why I combined these two, is to re-use the connection details. I could extract these to a separate function, but it would still need to be called twice. Or I'd have to expose the looked-up values somewhere more global (somewhere in `cnx`?). So in short, I haven't found a way to avoid calling the `getsockname()`-stuff twice). Can you give me some advice?

Reply:
This function being called for every connection, it makes sense to try and avoid doing it twice indeed. I've just pushed 3aa245e which should make it obvious what to do :-)

Reply:
Indeed; I've rebased on top of your current master, and used the provided `get_connection_desc()` function just like you do in the `log_connection()` function.